src/Utilities/SidreDataCollectionInline.hh

```diff
@@ -0,0 +1,531 @@
+#include "axom/sidre.hpp"
```

Comment:
We should add Comment headers to each of the new files explaining a little about what each one does